src/globus_sdk/experimental/session_error/session_error.py

```diff
@@ -0,0 +1,195 @@
+import copy
+import typing as t
+
+from globus_sdk.exc import GlobusError
+
+
+class GlobusSessionErrorAuthorizationParameters:
+    """
+    Represents authorization parameters that can be used to instruct a client
+    which additional authorizations are needed in order to complete a request.
+
+    :ivar session_message: A message to be displayed to the user.
+    :vartype session_message: str, optional
+
+    :ivar session_required_identities: A list of identities required for the
+        session.
+    :vartype session_required_identities: list of str, optional
+
+    :ivar session_required_policies: A list of policies required for the
+        session.
+    :vartype session_required_policies: list of str, optional
+
+    :ivar session_required_single_domain: A list of domains required for the
+        session.
+    :vartype session_required_single_domain: list of str, optional
+
+    :ivar session_required_mfa: Whether MFA is required for the session.
+    :vartype session_required_mfa: bool, optional
+
+    :ivar session_required_scopes: A list of scopes for which consent is required.
+    :vartype session_required_scopes: list of str, optional
+
+    :ivar extra_fields: A dictionary of additional fields that were provided. May
+        be used for forward/backward compatibility.
+    :vartype extra_fields: dict
+    """
+
+    SUPPORTED_FIELDS = {
+        "session_message": str,
+        "session_required_identities": list,
+        "session_required_policies": list,
+        "session_required_single_domain": list,
+        "session_required_mfa": bool,
+        "session_required_scopes": list,
+    }
+
+    def __init__(
+        self,
+        session_message: t.Optional[str] = None,
+        session_required_identities: t.Optional[t.List[str]] = None,
+        session_required_policies: t.Optional[t.List[str]] = None,
+        session_required_single_domain: t.Optional[t.List[str]] = None,
+        session_required_mfa: t.Optional[bool] = None,
+        session_required_scopes: t.Optional[t.List[str]] = None,
+        **kwargs: t.Any,
+    ):
```

Comment:
There are two changes I'd like us to make here -- consistent with the `**kwargs` and keyword-only argument notes recently added to the contrib doc.
```suggestion
        self,
        *,
        session_message: t.Optional[str] = None,
        session_required_identities: t.Optional[t.List[str]] = None,
        session_required_policies: t.Optional[t.List[str]] = None,
        session_required_single_domain: t.Optional[t.List[str]] = None,
        session_required_mfa: t.Optional[bool] = None,
        session_required_scopes: t.Optional[t.List[str]] = None,
        extra: dict[str, t.Any],
    ):
```